core/src/main/java/io/questdb/cairo/sql/AtomicBooleanCircuitBreaker.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.sql;
+
+import io.questdb.cairo.CairoException;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+// Circuit breaker that doesn't check network connection status or timeout and only allows cancelling statement via CANCEL QUERY command .
+public class AtomicBooleanCircuitBreaker implements SqlExecutionCircuitBreaker {
+    private final int throttle;
+    private AtomicBoolean cancelledFlag;
+    private int fd = -1;
+    private int testCount = 0;
+
+    public AtomicBooleanCircuitBreaker() {
+        this(0);
+        cancelledFlag = new AtomicBoolean(false);
+    }
+
+    public AtomicBooleanCircuitBreaker(int throttle) {
+        this.throttle = throttle;
+    }
+
+    public void cancel() {
+        cancelledFlag.set(true);
+    }
+
+    @Override
+    public int checkIfTripped(long millis, int fd) {
+        return isCancelled() ? STATE_CANCELLED : STATE_OK;
+    }
+
+    @Override
+    public boolean checkIfTripped() {
```

Comment:
it is confusing to have both `checkIfTripped()` and `isCancelled`. We should clean this API. Perhaps keep `checkIfTripped()` and also `int getState()` ?

Case in point, one usage looks like this:

![image](https://github.com/questdb/questdb/assets/7276403/4cddab54-c336-4aed-8fb0-616791b1abe7)

and the other:

![image](https://github.com/questdb/questdb/assets/7276403/ca1afb69-f502-4a95-af62-ea649e290664)

Reply:
Removed isCancelled() and added getState() methods in latest commit .